src/Processors/Formats/Impl/HiveTextRowInputFormat.cpp

```diff
@@ -0,0 +1,60 @@
+#include <Processors/Formats/Impl/HiveTextRowInputFormat.h>
+
+#if USE_HIVE
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NOT_IMPLEMENTED;
+}
+
+HiveTextRowInputFormat::HiveTextRowInputFormat(
+    const Block & header_, ReadBuffer & in_, const Params & params_, const FormatSettings & format_settings_)
+    : CSVRowInputFormat(header_, buf, params_, true, false, format_settings_)
+    , buf(in_)
+{
+}
+
+void HiveTextRowInputFormat::readPrefix()
```

Comment:
I would prefer to not override `readPrefix` and implement just `readNames` where we will read a row, rollback to it's start, resize input_field_names and return it as vector of names.

Reply:
ok